src/net/gateway-discovery.ts

```diff
@@ -256,6 +262,9 @@ export class GatewayDiscovery {
         // Sort DVote gateways by metrics
         healthyNodes.dvote.sort((a: IDVoteGateway, b: IDVoteGateway) => {
             switch (!!a && !!b) {
+                // Randomize if needed
+                case Math.random() < this.RANDOMIZE_SORTING:
```

Comment:
@marcvelmer I thought you would apply the randomization after the sorting. Because currently there is a case that a gateway is simply sorted  randomly.  I think Probably I am just commenting this because I liked the formula approach :-P

Reply:
How about using the weight as is was before? This way we can also use healt instead of using just one metric for ordering. Here is an example with random weight
`Math.floor(Math.random() * 100) * 20 / 100 a.performanceTime * 60/ 100 + a.health * 20 / 100 `

Reply:
But this would be only one solution for Dvote and not Web3. And on the other hand, with this criteria you used, you are giving more weight to those gateways with _more_ performanceTime

Reply:
For the first comment we could use a different formula for dvote and web3 if necessary or define the health of the web3 based on the `mostFrequentBlockNumber `

For the second comment we could use `1/a.performance time` and the result would be:
`Math.floor(Math.random() * 100) * 20 /( 100*a.performanceTime) * 60/ 100 + a.health * 20 / 100`